lib/server.js

```diff
@@ -98,6 +100,19 @@ function initSync(mediator, mbaasApi, datasetId, syncOptions) {
   return deferred.promise;
 }
 
+/**
+ * If the objects _id field is a Mongodb~ObjectID, convert it to a string.
+ *
+ * @param {String|Object} object - The object to check.
+ * @returns {String} - The original _id field or the converted Mongodb~ObjectID
+ */
+function getObjectId(object) {
```

Comment:
@david-martin I strongly disagree here. This is nothing to do with Raincatcher in any way.

If a developer was using the MongoDB client and doing the same thing, they would get the same unexpected behaviour from the sync framework.

In addition, if the document has an invalid `_id` field, then mongo will not create it. (e.g. when saving collisions) (i.e. you can't create a document with an _id field already in it) The sync framework need to deal with the fact that it uses mongodb under the hood.

If that is the case, then the sync framework needs to document that invalid JSON will produce unexpected results. (And possibly detect it here.)